src/transmission.js

```diff
@@ -284,79 +283,86 @@ export class Transmission {
     let batches = Object.keys(batchAgg.batches).map(k => batchAgg.batches[k]);
     eachPromise(batches, batch => {
       let url = urljoin(batch.apiHost, "/1/batch", batch.dataset);
-      let req = superagent.post(url);
-      if (this._proxy) {
-        req = proxy(req, this._proxy);
-      }
+      let postReq = superagent.post(url);
 
-      let { encoded, numEncoded } = batchAgg.encodeBatchEvents(batch.events);
-      return new Promise(resolve => {
-        // if we failed to encode any of the events, no point in sending anything to honeycomb
-        if (numEncoded === 0) {
-          this._responseCallback(
-            batch.events.map(ev => ({
-              metadata: ev.metadata,
-              error: ev.encodeError
+      const reqPromise = Promise.resolve(
+        this._proxy
+          ? import("superagent-proxy").then(({ default: proxy }) => ({
```

Comment:
will this make proxied requests any slower?

Reply:
`superagent-proxy` should still only be required a single time.  it will add an additional "tick" to each request thanks to the additional promise, but that shouldn't be more than a ms or two.